contracts/pool/concentrated/ConcentratedLiquidityPool.sol

```diff
@@ -123,10 +124,11 @@ contract ConcentratedLiquidityPool is IPool {
             _deployData,
             (address, address, uint24, uint160, uint24)
         );
-
+        if (TickMath.MIN_SQRT_RATIO <= price && price <= TickMath.MAX_SQRT_RATIO) revert BadPrice();
         if (_token0 == address(0)) revert ZeroAddress();
         if (_token0 == address(this)) revert InvalidToken();
         if (_token1 == address(this)) revert InvalidToken();
+        if (_token0 == _token1) revert InvalidToken();
```

Comment:
Should be >=

Reply:
No, we don't need to >, as the factory sorts the token already, however, it doesn't check if both tokens are same, this checks prevent that specific case.

Reply:
>= costs the same gas as ==

Reply:
Yes, but in this case, we don't need it, as the tokens will be sorted from the factory.